Source/game/gameloop.cpp

```diff
@@ -100,98 +92,88 @@ bool gameloop::run(int argc, char* argv[]) {
 
 		std::vector<unsigned int> indices = {};
 
+		/*----------------------------------------------------------------------------------*/
+
 		unsigned int VBO, VAO;
 		glGenBuffers(1, &VBO);
 		glGenVertexArrays(1, &VAO);
 
-		{
-			//Creates vertex/attribute object
-			std::unique_ptr<VertAttribObject> vertex_and_attrib_object = std::make_unique<VertAttribObject>(VAO, VBO);
-
-			vertex_and_attrib_object->bindVBO(vertices, sizeof(vertices), VBO);
-			vertex_and_attrib_object->bindVAO(VAO);
+		VertAttribObject vertex_and_attrib_object(VAO, VBO);
 
-			vertex_and_attrib_object->positionAttrib();
-			vertex_and_attrib_object->textureCoordAttrib();
+		vertex_and_attrib_object.bindVBO(vertices, sizeof(vertices), VBO);
+		vertex_and_attrib_object.bindVAO(VAO);
 
-			/*----------------------------------------------------------------------------------*/
+		vertex_and_attrib_object.positionAttrib();
+		vertex_and_attrib_object.textureCoordAttrib();
 
-			//TEXTURE 1
-			Texture our_texture1(environment::ResourcePath("Textures/MetalTexture1.jpg"));
-			//TEXTURE 2
-			Texture our_texture2(environment::ResourcePath("Textures/GraffitiTexture1.png"));
+		/*----------------------------------------------------------------------------------*/
 
-			/*----------------------------------------------------------------------------------*/
+		Texture texture1(environment::ResourcePath("Textures/metal_texture1.jpg"));
+		Texture texture2(environment::ResourcePath("Textures/graffiti_texture1.png"));
 
-			our_shader->use();
-			our_shader->setInt("texture2", 1); //Comment out to remove graffiti
+		/*----------------------------------------------------------------------------------*/
 
-			//Avoid cursor jump
-			glfwSetCursorPos(window, lastX, lastY);
+		shader.use();
+		shader.setInt("texture2", 1); // Comment out to remove graffiti
 
-			mouseProp(true, -90.0f, 0.0f, (800.0f / 2.0), (600.0 / 2.0), 45.0f);
+		// Avoid cursor jump
+		input::setCursorPos();
 
-			float time = 2.0f;
+		mouseProp(true, -90.0f, 0.0f, (800.0f / 2.0), (600.0 / 2.0), 45.0f);
 
-			//Input object
-			std::unique_ptr<PlayerCallback> test = std::make_unique<PlayerCallback>();
+		float time = 2.0f;
 
-			/*----------------------------------------------------------------------------------*/
+		PlayerCallback callbacks;
 
-			//Game loop
-			while (!glfwWindowShouldClose(window)) {
+		/*----------------------------------------------------------------------------------*/
 
-				test->update(delta_time);
+		// Game loop
+		while (context::window::isClosed(context::window::getWindow()) == false) {
 
-				//Creates camera object
-				Camera camera;
+			callbacks.update(delta_time);
 
-				camera.perFrameTimeLogic();
+			Camera camera;
```

Comment:
Construct the camera object outside the while loop, no need for it to be a new object every iteration.